core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala

```diff
@@ -2489,16 +2489,20 @@ class GroupCoordinatorTest {
     val rebalanceResult = staticMembersJoinAndRebalance(leaderInstanceId, followerInstanceId)
 
     val leaderNoMemberIdCommitOffsetResult = commitTransactionalOffsets(groupId, producerId, producerEpoch,
-      Map(tp -> offset), groupInstanceId = leaderInstanceId)
+      Map(tp -> offset), memberId = JoinGroupRequest.UNKNOWN_MEMBER_ID, groupInstanceId = leaderInstanceId)
     assertEquals(Errors.FENCED_INSTANCE_ID, leaderNoMemberIdCommitOffsetResult (tp))
 
+    val leaderInvalidMemberIdCommitOffsetResult = commitTransactionalOffsets(groupId, producerId, producerEpoch,
+      Map(tp -> offset), memberId = "invalid-member", groupInstanceId = leaderInstanceId)
+    assertEquals(Errors.FENCED_INSTANCE_ID, leaderInvalidMemberIdCommitOffsetResult (tp))
+
     val leaderCommitOffsetResult = commitTransactionalOffsets(groupId, producerId, producerEpoch,
       Map(tp -> offset), rebalanceResult.leaderId, leaderInstanceId)
     assertEquals(Errors.NONE, leaderCommitOffsetResult (tp))
   }
 
   @Test
-  def testTxnCommitOffsetWithUnknownMemberId(): Unit = {
+  def testTxnCommitOffsetWithInvalidMemberId(): Unit = {
```

Comment:
This is addressing the comments for separating valid and invalid scenario: https://github.com/apache/kafka/pull/7897#discussion_r366621880